test/unit/eslint-plugin-next/no-style-in-document.test.ts

```diff
@@ -0,0 +1,74 @@
+import rule from '@next/eslint-plugin-next/lib/rules/no-style-in-document'
+import { RuleTester } from 'eslint'
+;(RuleTester as any).setDefaultConfig({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+    ecmaFeatures: {
+      modules: true,
+      jsx: true,
+    },
+  },
+})
+const ruleTester = new RuleTester()
+
+ruleTester.run('no-style-in-document', rule, {
+  valid: [
```

Comment:
Could this add a regular `<style>` tag as that should be allowed?

Reply:
Yes. So basically the rule only checks for `<style>` with `jsx` attribute.

Reply:
I have added another valid testcase.